Automattic/jetpack
# JS Package: Add Critical CSS Gen

Fixes https://github.com/Automattic/boost-cloud/issues/267

## Proposed changes:

* Add critical css gen package to the monorepo:
  * Browser build is now in `build-browser` (was `dist` before);
  * Node build is now in `build-node` (was `lib` before);
* Update browser version to not include playwright/puppeteer code;
* Update node version to not include iframe code;
* ~Update Boost to use the new package instead of the external repo.~ This will be done in a follow-up PR.

### Other information:


## Jetpack product discussion

n/a

## Does this pull request change what data or activity we track or use?

no

## Testing instructions:

**Pre-requisites**

* Go to the root of the js package;
* Run `pnpm install`;
* Build the package using `pnpm build`;
* There should be no errors.

**Testing with the Boost plugin**

* Apply the changes from this commit 3f586d83adf54cef6a406a0b7ff5d1346428c597 (in reverse, the "removed" changes is what you need to add back to Boost);
* Setup free Boost;
* Install Boost dependencies using `jetpack install plugins/boost`;
* Build Boost using `jetpack build plugins/boost`;
* Try to generate critical css and make sure it's working.
